Could not read SSL key file when attempting to connect to Postgres DB with SSL using PKCS8 DER key file

### Describe the bug

- Attempted to connect to a GCP Postgres instance using SSL connection and Authenticating the client certificate, but receiving error about "Could not read SSL key file".
- Tried via uploading the key file via UI and putting the key file directly on server and neither worked.



### To Reproduce

1. Create a GCP CloudSQL Postgres instance with public IP (can have minimal resources), noting password set
2. Once created, go to the CloudSQL instance > Connections > Security, and enable Allow only SSL connections
3. In Connections > Security, click Create Client Certificate, and download the three files: `client-key.pem`, `client-cert.pem`, and `server-ca.pem`
4. In Connections > Networking, click Add a Network, and add the IP CIDR range including the public IP of your Metabase instance.
5. Try to add the database connection and check Use a secure connection (SSL), SSL Mode of `verify-ca`, and upload the `server-ca.pem`, select Authenticate client certificate, and upload the other two files.
6. Convert the `client-key.pem` to pk8 using `$ openssl pkcs8 -topk8 -inform PEM -outform DER -in client-key.pem -out client-key.pk8 -nocrypt`
7. Click Save and see the error:
![image](https://github.com/metabase/metabase/assets/8808703/1587572a-2f9a-466b-aed2-fc7921abb0d5)



### Expected behavior

- should be able to read key file

### Logs

```
[78559c23-4d6b-40b6-a36a-f91af9b8d99c] 2023-05-11T17:30:38-04:00 ERROR metabase.api.database Cannot connect to Database
clojure.lang.ExceptionInfo: Could not read SSL key file /tmp/metabase-secret_4980691877231097117.tmp. {:message "Could not read SSL key file /tmp/metabase-secret_4980691877231097117.tmp."}
	at metabase.driver.util$can_connect_with_details_QMARK_.invokeStatic(util.clj:159)
	at metabase.driver.util$can_connect_with_details_QMARK_.doInvoke(util.clj:132)
	at clojure.lang.RestFn.invoke(RestFn.java:442)
	at metabase.api.database$test_database_connection.invokeStatic(database.clj:677)
	at metabase.api.database$test_database_connection.doInvoke(database.clj:667)
	at clojure.lang.RestFn.invoke(RestFn.java:425)
	at metabase.api.database$fn__95595$test_connection_details__95600$fn__95601.invoke(database.clj:730)
	at metabase.api.database$fn__95595$test_connection_details__95600.invoke(database.clj:711)
	at metabase.api.database$fn__95628.invokeStatic(database.clj:749)
	at metabase.api.database$fn__95628.invoke(database.clj:735)
	at compojure.core$wrap_response$fn__42375.invoke(core.clj:160)
	at compojure.core$wrap_route_middleware$fn__42359.invoke(core.clj:132)
	at compojure.core$wrap_route_info$fn__42364.invoke(core.clj:139)
	at compojure.core$wrap_route_matches$fn__42368.invoke(core.clj:151)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__42368.invoke(core.clj:152)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387.invoke(core.clj:200)
	at metabase.server.middleware.auth$enforce_authentication$fn__87427.invoke(auth.clj:17)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387.invoke(core.clj:200)
	at compojure.core$make_context$handler__42415.invoke(core.clj:290)
	at compojure.core$make_context$fn__42419.invoke(core.clj:300)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:199)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:199)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387.invoke(core.clj:200)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:199)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387.invoke(core.clj:200)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387.invoke(core.clj:200)
	at metabase.api.routes$fn__99241$fn__99242.invoke(routes.clj:62)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387.invoke(core.clj:200)
	at clojure.lang.AFn.applyToHelper(AFn.java:160)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:662)
	at metabase.server.routes$fn__99402$fn__99403.doInvoke(routes.clj:70)
	at clojure.lang.RestFn.invoke(RestFn.java:436)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387.invoke(core.clj:200)
	at compojure.core$make_context$handler__42415.invoke(core.clj:290)
	at compojure.core$make_context$fn__42419.invoke(core.clj:300)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__42368.invoke(core.clj:153)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__42368.invoke(core.clj:153)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__42368.invoke(core.clj:153)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:199)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387.invoke(core.clj:200)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387.invoke(core.clj:200)
	at metabase.server.middleware.exceptions$catch_uncaught_exceptions$fn__95938.invoke(exceptions.clj:103)
	at metabase.server.middleware.exceptions$catch_api_exceptions$fn__95935.invoke(exceptions.clj:91)
	at metabase.server.middleware.log$log_api_call$fn__102699$fn__102700$fn__102701.invoke(log.clj:216)
	at metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info.invokeStatic(diagnostic.clj:18)
	at metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info.invoke(diagnostic.clj:12)
	at metabase.server.middleware.log$log_api_call$fn__102699$fn__102700.invoke(log.clj:208)
	at toucan2.execute$do_with_call_counts.invokeStatic(execute.clj:112)
	at toucan2.execute$do_with_call_counts.invoke(execute.clj:103)
	at metabase.server.middleware.log$log_api_call$fn__102699.invoke(log.clj:207)
	at metabase.server.middleware.browser_cookie$ensure_browser_id_cookie$fn__109422.invoke(browser_cookie.clj:42)
	at metabase.server.middleware.security$add_security_headers$fn__80256.invoke(security.clj:153)
	at metabase.server.middleware.json$wrap_json_body$fn__106674.invoke(json.clj:67)
	at metabase.server.middleware.offset_paging$handle_paging$fn__80280.invoke(offset_paging.clj:45)
	at metabase.server.middleware.json$wrap_streamed_json_response$fn__106692.invoke(json.clj:103)
	at ring.middleware.keyword_params$wrap_keyword_params$fn__109689.invoke(keyword_params.clj:55)
	at ring.middleware.params$wrap_params$fn__109708.invoke(params.clj:77)
	at metabase.server.middleware.misc$maybe_set_site_url$fn__50479.invoke(misc.clj:61)
	at metabase.server.middleware.session$reset_session_timeout$fn__63770.invoke(session.clj:430)
	at metabase.server.middleware.session$bind_current_user$fn__63740$fn__63741.invoke(session.clj:336)
	at metabase.server.middleware.session$do_with_current_user.invokeStatic(session.clj:315)
	at metabase.server.middleware.session$do_with_current_user.invoke(session.clj:304)
	at metabase.server.middleware.session$bind_current_user$fn__63740.invoke(session.clj:335)
	at metabase.server.middleware.session$wrap_current_user_info$fn__63722.invoke(session.clj:285)
	at metabase.server.middleware.session$wrap_session_id$fn__63705.invoke(session.clj:217)
	at metabase.server.middleware.auth$wrap_api_key$fn__87435.invoke(auth.clj:30)
	at ring.middleware.cookies$wrap_cookies$fn__109609.invoke(cookies.clj:216)
	at metabase.server.middleware.misc$add_content_type$fn__50461.invoke(misc.clj:29)
	at metabase.server.middleware.misc$disable_streaming_buffering$fn__50487.invoke(misc.clj:78)
	at ring.middleware.gzip$wrap_gzip$fn__109651.invoke(gzip.clj:86)
	at metabase.server.middleware.misc$bind_request$fn__50490.invoke(misc.clj:95)
	at metabase.server.middleware.ssl$redirect_to_https_middleware$fn__109438.invoke(ssl.clj:41)
	at metabase.server$async_proxy_handler$fn__101465.invoke(server.clj:77)
	at metabase.server.proxy$org.eclipse.jetty.server.handler.AbstractHandler$ff19274a.handle(Unknown Source)
	at org.eclipse.jetty.server.handler.StatisticsHandler.handle(StatisticsHandler.java:173)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122)
	at org.eclipse.jetty.server.Server.handle(Server.java:563)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$0(HttpChannel.java:505)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:762)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:497)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:282)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:314)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:100)
	at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:53)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask(AdaptiveExecutionStrategy.java:416)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask(AdaptiveExecutionStrategy.java:385)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce(AdaptiveExecutionStrategy.java:272)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.lambda$new$0(AdaptiveExecutionStrategy.java:140)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:411)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:936)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1080)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: org.postgresql.util.PSQLException: Could not read SSL key file /tmp/metabase-secret_4980691877231097117.tmp.
	at org.postgresql.ssl.LazyKeyManager.getPrivateKey(LazyKeyManager.java:284)
	at java.base/sun.security.ssl.AbstractKeyManagerWrapper.getPrivateKey(Unknown Source)
	at java.base/sun.security.ssl.X509Authentication$X509PossessionGenerator.createClientPossession(Unknown Source)
	at java.base/sun.security.ssl.X509Authentication$X509PossessionGenerator.createPossession(Unknown Source)
	at java.base/sun.security.ssl.X509Authentication.createPossession(Unknown Source)
	at java.base/sun.security.ssl.CertificateMessage$T13CertificateProducer.choosePossession(Unknown Source)
	at java.base/sun.security.ssl.CertificateMessage$T13CertificateProducer.onProduceCertificate(Unknown Source)
	at java.base/sun.security.ssl.CertificateMessage$T13CertificateProducer.produce(Unknown Source)
	at java.base/sun.security.ssl.SSLHandshake.produce(Unknown Source)
	at java.base/sun.security.ssl.Finished$T13FinishedConsumer.onConsumeFinished(Unknown Source)
	at java.base/sun.security.ssl.Finished$T13FinishedConsumer.consume(Unknown Source)
	at java.base/sun.security.ssl.SSLHandshake.consume(Unknown Source)
	at java.base/sun.security.ssl.HandshakeContext.dispatch(Unknown Source)
	at java.base/sun.security.ssl.HandshakeContext.dispatch(Unknown Source)
	at java.base/sun.security.ssl.TransportContext.dispatch(Unknown Source)
	at java.base/sun.security.ssl.SSLTransport.decode(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.decode(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(Unknown Source)
	at org.postgresql.ssl.MakeSSL.convert(MakeSSL.java:41)
	at org.postgresql.core.v3.ConnectionFactoryImpl.enableSSL(ConnectionFactoryImpl.java:584)
	at org.postgresql.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:168)
	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:235)
	at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:49)
	at org.postgresql.jdbc.PgConnection.<init>(PgConnection.java:247)
	at org.postgresql.Driver.makeConnection(Driver.java:434)
	at org.postgresql.Driver.connect(Driver.java:291)
	at java.sql/java.sql.DriverManager.getConnection(Unknown Source)
	at java.sql/java.sql.DriverManager.getConnection(Unknown Source)
	at clojure.java.jdbc$get_driver_connection.invokeStatic(jdbc.clj:271)
	at clojure.java.jdbc$get_driver_connection.invoke(jdbc.clj:250)
	at clojure.java.jdbc$get_connection.invokeStatic(jdbc.clj:411)
	at clojure.java.jdbc$get_connection.invoke(jdbc.clj:274)
	at clojure.java.jdbc$db_query_with_resultset_STAR_.invokeStatic(jdbc.clj:1111)
	at clojure.java.jdbc$db_query_with_resultset_STAR_.invoke(jdbc.clj:1093)
	at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1182)
	at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
	at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1160)
	at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
	at metabase.driver.sql_jdbc.connection$can_connect_with_spec_QMARK_.invokeStatic(connection.clj:302)
	at metabase.driver.sql_jdbc.connection$can_connect_with_spec_QMARK_.invoke(connection.clj:299)
	at metabase.driver.sql_jdbc.connection$can_connect_QMARK_$fn__50099.invoke(connection.clj:311)
	at metabase.driver.sql_jdbc.connection$do_with_connection_spec_for_testing_connection$fn__50081.invoke(connection.clj:287)
	at metabase.util.ssh$do_with_ssh_tunnel.invokeStatic(ssh.clj:141)
	at metabase.util.ssh$do_with_ssh_tunnel.invoke(ssh.clj:130)
	at metabase.driver.sql_jdbc.connection$do_with_connection_spec_for_testing_connection.invokeStatic(connection.clj:285)
	at metabase.driver.sql_jdbc.connection$do_with_connection_spec_for_testing_connection.invoke(connection.clj:281)
	at metabase.driver.sql_jdbc.connection$can_connect_QMARK_.invokeStatic(connection.clj:310)
	at metabase.driver.sql_jdbc.connection$can_connect_QMARK_.invoke(connection.clj:306)
	at metabase.driver.sql_jdbc$fn__111370.invokeStatic(sql_jdbc.clj:43)
	at metabase.driver.sql_jdbc$fn__111370.invoke(sql_jdbc.clj:41)
	at clojure.lang.MultiFn.invoke(MultiFn.java:234)
	at metabase.driver.util$can_connect_with_details_QMARK_$fn__46839.invoke(util.clj:144)
	at clojure.core$binding_conveyor_fn$fn__5823.invoke(core.clj:2047)
	at clojure.lang.AFn.call(AFn.java:18)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	... 1 more
Caused by: java.io.IOException: DerInputStream.getLength(): lengthTag=111, too big.
	at java.base/sun.security.util.DerInputStream.getLength(Unknown Source)
	at java.base/sun.security.util.DerValue.init(Unknown Source)
	at java.base/sun.security.util.DerValue.<init>(Unknown Source)
	at java.base/sun.security.util.DerValue.<init>(Unknown Source)
	at java.base/sun.security.util.DerValue.wrap(Unknown Source)
	at java.base/sun.security.util.DerValue.wrap(Unknown Source)
	at java.base/javax.crypto.EncryptedPrivateKeyInfo.<init>(Unknown Source)
	at org.postgresql.ssl.LazyKeyManager.getPrivateKey(LazyKeyManager.java:236)
	... 59 more
[78559c23-4d6b-40b6-a36a-f91af9b8d99c] 2023-05-11T17:30:38-04:00 DEBUG metabase.server.middleware.log POST /api/database 400 286.3 ms (0 DB calls) 
{:message "Could not read SSL key file /tmp/metabase-secret_4980691877231097117.tmp."}
```

### Information about your Metabase installation

- Metabase v1.46.2


### Severity
- Unable to create verified connection with standard supported database

### Additional context

_No response_

Forgot to convert to PKCS8 DER, but even after doing that, still getting an issue loading key:
```
$ openssl pkcs8 -topk8 -inform PEM -outform DER -in client-key.pem -out client-key.pk8 -nocrypt
```

After uploading that file, still getting `Could not read SSL key file`
![image](https://github.com/metabase/metabase/assets/8808703/be047eec-9910-47fa-b6f6-0bf6a322e2c7)



```
[78559c23-4d6b-40b6-a36a-f91af9b8d99c] 2023-05-11T17:30:38-04:00 ERROR metabase.api.database Cannot connect to Database
clojure.lang.ExceptionInfo: Could not read SSL key file /tmp/metabase-secret_4980691877231097117.tmp. {:message "Could not read SSL key file /tmp/metabase-secret_4980691877231097117.tmp."}
	at metabase.driver.util$can_connect_with_details_QMARK_.invokeStatic(util.clj:159)
	at metabase.driver.util$can_connect_with_details_QMARK_.doInvoke(util.clj:132)
	at clojure.lang.RestFn.invoke(RestFn.java:442)
	at metabase.api.database$test_database_connection.invokeStatic(database.clj:677)
	at metabase.api.database$test_database_connection.doInvoke(database.clj:667)
	at clojure.lang.RestFn.invoke(RestFn.java:425)
	at metabase.api.database$fn__95595$test_connection_details__95600$fn__95601.invoke(database.clj:730)
	at metabase.api.database$fn__95595$test_connection_details__95600.invoke(database.clj:711)
	at metabase.api.database$fn__95628.invokeStatic(database.clj:749)
	at metabase.api.database$fn__95628.invoke(database.clj:735)
	at compojure.core$wrap_response$fn__42375.invoke(core.clj:160)
	at compojure.core$wrap_route_middleware$fn__42359.invoke(core.clj:132)
	at compojure.core$wrap_route_info$fn__42364.invoke(core.clj:139)
	at compojure.core$wrap_route_matches$fn__42368.invoke(core.clj:151)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__42368.invoke(core.clj:152)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387.invoke(core.clj:200)
	at metabase.server.middleware.auth$enforce_authentication$fn__87427.invoke(auth.clj:17)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387.invoke(core.clj:200)
	at compojure.core$make_context$handler__42415.invoke(core.clj:290)
	at compojure.core$make_context$fn__42419.invoke(core.clj:300)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:199)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:199)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387.invoke(core.clj:200)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:199)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387.invoke(core.clj:200)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387.invoke(core.clj:200)
	at metabase.api.routes$fn__99241$fn__99242.invoke(routes.clj:62)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387.invoke(core.clj:200)
	at clojure.lang.AFn.applyToHelper(AFn.java:160)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:662)
	at metabase.server.routes$fn__99402$fn__99403.doInvoke(routes.clj:70)
	at clojure.lang.RestFn.invoke(RestFn.java:436)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387.invoke(core.clj:200)
	at compojure.core$make_context$handler__42415.invoke(core.clj:290)
	at compojure.core$make_context$fn__42419.invoke(core.clj:300)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__42368.invoke(core.clj:153)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__42368.invoke(core.clj:153)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$wrap_route_matches$fn__42368.invoke(core.clj:153)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:199)
	at compojure.core$routes$fn__42387$f__42388$respond_SINGLEQUOTE___42389.invoke(core.clj:197)
	at compojure.core$make_context$fn__42419.invoke(core.clj:301)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387.invoke(core.clj:200)
	at compojure.core$routes$fn__42387$f__42388.invoke(core.clj:198)
	at compojure.core$routes$fn__42387.invoke(core.clj:200)
	at metabase.server.middleware.exceptions$catch_uncaught_exceptions$fn__95938.invoke(exceptions.clj:103)
	at metabase.server.middleware.exceptions$catch_api_exceptions$fn__95935.invoke(exceptions.clj:91)
	at metabase.server.middleware.log$log_api_call$fn__102699$fn__102700$fn__102701.invoke(log.clj:216)
	at metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info.invokeStatic(diagnostic.clj:18)
	at metabase.driver.sql_jdbc.execute.diagnostic$do_with_diagnostic_info.invoke(diagnostic.clj:12)
	at metabase.server.middleware.log$log_api_call$fn__102699$fn__102700.invoke(log.clj:208)
	at toucan2.execute$do_with_call_counts.invokeStatic(execute.clj:112)
	at toucan2.execute$do_with_call_counts.invoke(execute.clj:103)
	at metabase.server.middleware.log$log_api_call$fn__102699.invoke(log.clj:207)
	at metabase.server.middleware.browser_cookie$ensure_browser_id_cookie$fn__109422.invoke(browser_cookie.clj:42)
	at metabase.server.middleware.security$add_security_headers$fn__80256.invoke(security.clj:153)
	at metabase.server.middleware.json$wrap_json_body$fn__106674.invoke(json.clj:67)
	at metabase.server.middleware.offset_paging$handle_paging$fn__80280.invoke(offset_paging.clj:45)
	at metabase.server.middleware.json$wrap_streamed_json_response$fn__106692.invoke(json.clj:103)
	at ring.middleware.keyword_params$wrap_keyword_params$fn__109689.invoke(keyword_params.clj:55)
	at ring.middleware.params$wrap_params$fn__109708.invoke(params.clj:77)
	at metabase.server.middleware.misc$maybe_set_site_url$fn__50479.invoke(misc.clj:61)
	at metabase.server.middleware.session$reset_session_timeout$fn__63770.invoke(session.clj:430)
	at metabase.server.middleware.session$bind_current_user$fn__63740$fn__63741.invoke(session.clj:336)
	at metabase.server.middleware.session$do_with_current_user.invokeStatic(session.clj:315)
	at metabase.server.middleware.session$do_with_current_user.invoke(session.clj:304)
	at metabase.server.middleware.session$bind_current_user$fn__63740.invoke(session.clj:335)
	at metabase.server.middleware.session$wrap_current_user_info$fn__63722.invoke(session.clj:285)
	at metabase.server.middleware.session$wrap_session_id$fn__63705.invoke(session.clj:217)
	at metabase.server.middleware.auth$wrap_api_key$fn__87435.invoke(auth.clj:30)
	at ring.middleware.cookies$wrap_cookies$fn__109609.invoke(cookies.clj:216)
	at metabase.server.middleware.misc$add_content_type$fn__50461.invoke(misc.clj:29)
	at metabase.server.middleware.misc$disable_streaming_buffering$fn__50487.invoke(misc.clj:78)
	at ring.middleware.gzip$wrap_gzip$fn__109651.invoke(gzip.clj:86)
	at metabase.server.middleware.misc$bind_request$fn__50490.invoke(misc.clj:95)
	at metabase.server.middleware.ssl$redirect_to_https_middleware$fn__109438.invoke(ssl.clj:41)
	at metabase.server$async_proxy_handler$fn__101465.invoke(server.clj:77)
	at metabase.server.proxy$org.eclipse.jetty.server.handler.AbstractHandler$ff19274a.handle(Unknown Source)
	at org.eclipse.jetty.server.handler.StatisticsHandler.handle(StatisticsHandler.java:173)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:122)
	at org.eclipse.jetty.server.Server.handle(Server.java:563)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$0(HttpChannel.java:505)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:762)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:497)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:282)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:314)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:100)
	at org.eclipse.jetty.io.SelectableChannelEndPoint$1.run(SelectableChannelEndPoint.java:53)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.runTask(AdaptiveExecutionStrategy.java:416)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.consumeTask(AdaptiveExecutionStrategy.java:385)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.tryProduce(AdaptiveExecutionStrategy.java:272)
	at org.eclipse.jetty.util.thread.strategy.AdaptiveExecutionStrategy.lambda$new$0(AdaptiveExecutionStrategy.java:140)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:411)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:936)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1080)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: org.postgresql.util.PSQLException: Could not read SSL key file /tmp/metabase-secret_4980691877231097117.tmp.
	at org.postgresql.ssl.LazyKeyManager.getPrivateKey(LazyKeyManager.java:284)
	at java.base/sun.security.ssl.AbstractKeyManagerWrapper.getPrivateKey(Unknown Source)
	at java.base/sun.security.ssl.X509Authentication$X509PossessionGenerator.createClientPossession(Unknown Source)
	at java.base/sun.security.ssl.X509Authentication$X509PossessionGenerator.createPossession(Unknown Source)
	at java.base/sun.security.ssl.X509Authentication.createPossession(Unknown Source)
	at java.base/sun.security.ssl.CertificateMessage$T13CertificateProducer.choosePossession(Unknown Source)
	at java.base/sun.security.ssl.CertificateMessage$T13CertificateProducer.onProduceCertificate(Unknown Source)
	at java.base/sun.security.ssl.CertificateMessage$T13CertificateProducer.produce(Unknown Source)
	at java.base/sun.security.ssl.SSLHandshake.produce(Unknown Source)
	at java.base/sun.security.ssl.Finished$T13FinishedConsumer.onConsumeFinished(Unknown Source)
	at java.base/sun.security.ssl.Finished$T13FinishedConsumer.consume(Unknown Source)
	at java.base/sun.security.ssl.SSLHandshake.consume(Unknown Source)
	at java.base/sun.security.ssl.HandshakeContext.dispatch(Unknown Source)
	at java.base/sun.security.ssl.HandshakeContext.dispatch(Unknown Source)
	at java.base/sun.security.ssl.TransportContext.dispatch(Unknown Source)
	at java.base/sun.security.ssl.SSLTransport.decode(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.decode(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(Unknown Source)
	at org.postgresql.ssl.MakeSSL.convert(MakeSSL.java:41)
	at org.postgresql.core.v3.ConnectionFactoryImpl.enableSSL(ConnectionFactoryImpl.java:584)
	at org.postgresql.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:168)
	at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:235)
	at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:49)
	at org.postgresql.jdbc.PgConnection.<init>(PgConnection.java:247)
	at org.postgresql.Driver.makeConnection(Driver.java:434)
	at org.postgresql.Driver.connect(Driver.java:291)
	at java.sql/java.sql.DriverManager.getConnection(Unknown Source)
	at java.sql/java.sql.DriverManager.getConnection(Unknown Source)
	at clojure.java.jdbc$get_driver_connection.invokeStatic(jdbc.clj:271)
	at clojure.java.jdbc$get_driver_connection.invoke(jdbc.clj:250)
	at clojure.java.jdbc$get_connection.invokeStatic(jdbc.clj:411)
	at clojure.java.jdbc$get_connection.invoke(jdbc.clj:274)
	at clojure.java.jdbc$db_query_with_resultset_STAR_.invokeStatic(jdbc.clj:1111)
	at clojure.java.jdbc$db_query_with_resultset_STAR_.invoke(jdbc.clj:1093)
	at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1182)
	at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
	at clojure.java.jdbc$query.invokeStatic(jdbc.clj:1160)
	at clojure.java.jdbc$query.invoke(jdbc.clj:1144)
	at metabase.driver.sql_jdbc.connection$can_connect_with_spec_QMARK_.invokeStatic(connection.clj:302)
	at metabase.driver.sql_jdbc.connection$can_connect_with_spec_QMARK_.invoke(connection.clj:299)
	at metabase.driver.sql_jdbc.connection$can_connect_QMARK_$fn__50099.invoke(connection.clj:311)
	at metabase.driver.sql_jdbc.connection$do_with_connection_spec_for_testing_connection$fn__50081.invoke(connection.clj:287)
	at metabase.util.ssh$do_with_ssh_tunnel.invokeStatic(ssh.clj:141)
	at metabase.util.ssh$do_with_ssh_tunnel.invoke(ssh.clj:130)
	at metabase.driver.sql_jdbc.connection$do_with_connection_spec_for_testing_connection.invokeStatic(connection.clj:285)
	at metabase.driver.sql_jdbc.connection$do_with_connection_spec_for_testing_connection.invoke(connection.clj:281)
	at metabase.driver.sql_jdbc.connection$can_connect_QMARK_.invokeStatic(connection.clj:310)
	at metabase.driver.sql_jdbc.connection$can_connect_QMARK_.invoke(connection.clj:306)
	at metabase.driver.sql_jdbc$fn__111370.invokeStatic(sql_jdbc.clj:43)
	at metabase.driver.sql_jdbc$fn__111370.invoke(sql_jdbc.clj:41)
	at clojure.lang.MultiFn.invoke(MultiFn.java:234)
	at metabase.driver.util$can_connect_with_details_QMARK_$fn__46839.invoke(util.clj:144)
	at clojure.core$binding_conveyor_fn$fn__5823.invoke(core.clj:2047)
	at clojure.lang.AFn.call(AFn.java:18)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	... 1 more
Caused by: java.io.IOException: DerInputStream.getLength(): lengthTag=111, too big.
	at java.base/sun.security.util.DerInputStream.getLength(Unknown Source)
	at java.base/sun.security.util.DerValue.init(Unknown Source)
	at java.base/sun.security.util.DerValue.<init>(Unknown Source)
	at java.base/sun.security.util.DerValue.<init>(Unknown Source)
	at java.base/sun.security.util.DerValue.wrap(Unknown Source)
	at java.base/sun.security.util.DerValue.wrap(Unknown Source)
	at java.base/javax.crypto.EncryptedPrivateKeyInfo.<init>(Unknown Source)
	at org.postgresql.ssl.LazyKeyManager.getPrivateKey(LazyKeyManager.java:236)
	... 59 more
[78559c23-4d6b-40b6-a36a-f91af9b8d99c] 2023-05-11T17:30:38-04:00 DEBUG metabase.server.middleware.log POST /api/database 400 286.3 ms (0 DB calls) 
{:message "Could not read SSL key file /tmp/metabase-secret_4980691877231097117.tmp."}
```

--- a/frontend/src/metabase/databases/components/DatabaseDetailField/DatabaseDetailField.tsx
+++ b/frontend/src/metabase/databases/components/DatabaseDetailField/DatabaseDetailField.tsx
@@ -62,7 +62,7 @@ const getFieldProps = (field: EngineField, override?: EngineFieldOverride) => {
     title: override?.title ?? field["display-name"],
     description: override?.description ?? field.description,
     placeholder: placeholder != null ? String(placeholder) : undefined,
-    encoding: override ? null : field["treat-before-posting"],
+    encoding: field["treat-before-posting"],
   };
 };
 

--- a/frontend/src/metabase/databases/components/DatabaseDetailField/DatabaseDetailField.tsx
+++ b/frontend/src/metabase/databases/components/DatabaseDetailField/DatabaseDetailField.tsx
@@ -62,6 +62,7 @@ const getFieldProps = (field: EngineField, override?: EngineFieldOverride) => {
     title: override?.title ?? field["display-name"],
     description: override?.description ?? field.description,
     placeholder: placeholder != null ? String(placeholder) : undefined,
+    encoding: override ? null : field["treat-before-posting"],
   };
 };
 

